fiftyone/core/sample.py

```diff
@@ -57,13 +57,15 @@ def __getitem__(self, field_name):
         if self.media_type == fomm.VIDEO and fofu.is_frame_number(field_name):
             return self.frames[field_name]
 
+        self._check_reserved_keywords([field_name])
         return super().__getitem__(field_name)
 
     def __setitem__(self, field_name, value):
         if self.media_type == fomm.VIDEO and fofu.is_frame_number(field_name):
             self.frames[field_name] = value
             return
 
+        self._check_reserved_keywords([field_name])
```

Comment:
The `Sample` class inherits from `fiftyone.core.document._Document`:
https://github.com/voxel51/fiftyone/blob/75acb79de8362efaf02d79fe2697e5338106606c/fiftyone/core/document.py#L19

which maintains a `_doc` property that uses the `fiftyone.core.odm.document.MongoEngineBaseDocument` interface to actually store the field values:
https://github.com/voxel51/fiftyone/blob/75acb79de8362efaf02d79fe2697e5338106606c/fiftyone/core/odm/document.py#L319

I believe that's the source of the `pk` issue, so it would be the appropriate place to do the reserved keyword check, as this same issue would apply to all users of this class, including other public-facing classes like `fiftyone.core.frame.Frame` that aren't currently covered by this PR.